bundles.md

```diff
@@ -0,0 +1,131 @@
+# Attestation Bundle
+in-toto attestations support multiple different predicate types with the idea that
+users will want to propagage multiple assertions about a single software package
+(Provenance, Code Review, Test Result, vuln scan, ...).
+
+Attestation Bundles provides a canonical method of storing those attestations in
+a single place.
+
+The attestations stored in a bundle may all come from the same source, signed by
+the same key, or they come from multiple sources, signed by different keys.  New
+attestations would simply be added to the existing bundle.
+
+**NOTE**: Attestation Bundles themselves are **not authenticated** instead each
+individual attestation is authenticated
+([using DSSE](https://github.com/secure-systems-lab/dsse)). As such, an attacker
+might be able to _delete_ an attestation without being detected.  Predicates that
+follow [the monotonic principle](spec/README.md#parsing-rules) should not have any
+issues with this behavior.
+
+## Data structure
+
+Attestation Bundles use [JSON Lines](https://jsonlines.org/) to store multiple
+[DSSEs](https://github.com/secure-systems-lab/dsse).
+
+*  Each line MUST contain a _single_ DSSE.
+*  Each line MAY contain any DSSE `payloadType`
+*  in-toto attestations (`payloadType` == `application/vnd.in-toto+json`) MAY use
+   any ` _type`/`predicateType`
+*  [in-toto Statements](https://github.com/in-toto/attestation/blob/main/spec/README.md#statement)
+   in a Bundle MAY reference different Subjects
+*  Consumers MUST ignore any attestations whose `payloadType`, `_type_`, `predicateType`
+   they do not understand.
+
+Example:
+```
+{ "payloadType": "application/vnd.in-toto+json", "payload": "...", "signatures": [...] }
+{ "payloadType": "application/vnd.in-toto+json", "payload": "...", "signatures": [...] }
+{ "payloadType": "application/vnd.foo+xml", "payload": "...", "signatures": [...] }
+```
+
+## File naming convention
+
+* Attestation Bundles SHOULD use the `.intoto.jsonl` extension.
+* Bundles that concern a single artifact SHOULD name the bundle file
+  `<artifact filename>.intoto.jsonl`.
+* Bundles that concern multiple artifacts SHOULD name the bundle file
+  `build.intoto.jsonl`.
+
+## Example Use Case
```

Comment:
nit: I think this section is valuable, but it is pretty long relative to the rest of the doc. Should we make it significantly shorter?

Reply:
Any thoughts on how?  I could just cut some sections.

Reply:
It's also kinda repetitive. Maybe we can take out the redundant example above.

Reply:
Some examples here can definitely be cut, they look pretty much the same at first glance. We should insert different examples covering the different combinations of SHOULDs and MAYs in the data structure above.

Reply:
I made these changes:

* Removed 'Testing' section since it was redundant with others.
* Updated the vuln scanning section to use a different payloadType to demonstrate that not everything needs to be an in-toto statement.
* Updated the deployment section to indicate that the policy engine skips attestations it doesn't understand (in this case SPDX, but it could easily be anything else).
* Removed example from the data structure section.

Reply:
LGTM